### ISSUE TITLE
where is the lane vis output tensor in paper?

### ISSUE THREAD

**Issue by zimurui** [OpenPerceptionX/PersFormer_3DLane#36]
thank you for your share, but I find that the github code  is not consistant with paper. 

such as,  "Unifying anchor design in 2D and 3D. We first put curated anchors (red) in the BEV space (left), then project them to the front view (right). Offset xi k and ui k (dashed line) are predicted to match ground truth (yellow and green) to anchors. The correspondence is thus built, and features are optimized together“

the code predicting lane length using the laneatt old method, which predict the start ration, but paper uses the lane visible for each row??

**Comment by ilnehc** [comment]
@zimurui Hi, could you elaborate more cause I did not fully understand your question. LaneATT predicts the length of lanes directly while our 2D branch predicts each row with visibility to cope with the final length.

**Comment by zimurui** [comment]
> @zimurui Hi, could you elaborate more cause I did not fully understand your question. LaneATT predicts the length of lanes directly while our 2D branch predicts each row with visibility to cope with the final length.

Thank your replay.
my question is that I fail to find the codes about "predicts each row with visibility"?  is that in "models/networks/Lane2D.py"? But I found the 2D lane detection head is same as the laneatt,  which predict the lane length by codes (copying from your Lane2D.py script):

"reg_proposals = torch.zeros((*cls_logits.shape[:2], self.num_category + 2 + 2 * self.n_offsets), device=batch_features.device)
  reg_proposals += self.anchors
  reg_proposals[:, :, :self.num_category] = cls_logits
   reg_proposals[:, :, self.num_category+2:] += reg
   # Apply nms
  # proposals_list = self.nms(reg_proposals, attention_matrix, nms_thres, nms_topk, conf_threshold, eval=eval)
  proposals_list = []
  for proposals, att_matrix in zip(reg_proposals, attention_matrix):
            anchor_inds = torch.arange(reg_proposals.shape[1], device=proposals.device)
            proposals_list.append((proposals, self.anchors, att_matrix, anchor_inds))
  return proposals_list
"
so,  whereis th each row visibility in 2D branch about lane length prediction?

**Comment by ilnehc** [comment]
@zimurui Yes, it is different from the original LaneATT and we did not try it. We design this to make 2D/3D branches consistent at the very beginning. Intuitively, the critical problem is that LaneATT has too many anchors to keep them both in 2D and 3D.

> Add more a question, could you please provide some trained weights?

Sorry for that. We could not provide pretrained weights in accordance with Waymo license.
